### DISCUSSION TITLE
Unwanted warning for dynamic input properties to actions

### DISCUSSION THREAD

**Discussion by joshuanapoli** [community/community#25387] [Actions]
<p>Actions recently started emitting a warning when giving input properties of actions that are not listed in the action’s inputs. For example:</p>
<pre><code class="lang-auto">##[warning]Unexpected input 'service', valid inputs are ['file']
</code></pre>
<p><a href="https://github.com/CumulusDS/get-yaml-paths-action" rel="nofollow noopener">My action</a>’s inputs are dynamic. I would like to define a specific required property, but allow users to give other input properties without emitting a warning.</p>
<p>How can I accomplish this?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/joshuanapoli">@joshuanapoli</a> ,<br>
The warning is showing as expected.  Currently, there is no method to disable the warning. Would you mind sharing your idea in the [Feedback form for GitHub Actions] ? (<a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">https://support.github.com/contact/feedback?contact[category]=actions</a>).<br>
As a workaround , I would recommend you using environment variable to simulate input. You need to use upper cases and add ‘INPUT_’ as the prefix of environment variable names.<br>
For example:</p>
<pre><code class="lang-auto">- uses: CumulusDS/get-yaml-paths-action@v0.1.0
  with: 
    file: file.yml
  env: 
    INPUT_BAR: foo.bar
    INPUT_PROVIDERSTAGE: provider.stage
    
</code></pre>

**Comment by joshuanapoli** [comment]
<p>Thanks. I submitted a support feedback, as you recommended. Note that the problem also affects the octokit Request Action: <a href="https://github.com/octokit/request-action/issues/26" rel="nofollow noopener">https://github.com/octokit/request-action/issues/26</a></p>

**Comment by Yanjingzhu** [comment]
<p>Thank you for your cooperation. The warning is added to the latest version of GitHub hosted runner.  <a href="https://github.com/actions/runner/issues/514" rel="nofollow noopener">There</a> is the issue I created in actions/runner repo reporting this input warning issue. You could keep an eye on it for any updates.</p>
